test/jdk/java/net/httpclient/http2/IdleConnectionTimeoutTest.java

```diff
@@ -0,0 +1,161 @@
+ * @test
+ * @bug 8263031
+ * @summary Tests that the HttpClient can correctly receive a Push Promise
+ *          Frame with the END_HEADERS flag unset followed by one or more
+ *          Continuation Frames.
+ * @library /test/lib server
+ * @build jdk.test.lib.net.SimpleSSLContext
+ * @modules java.base/sun.net.www.http
+ *          java.net.http/jdk.internal.net.http.common
+ *          java.net.http/jdk.internal.net.http.frame
+ *          java.net.http/jdk.internal.net.http.hpack
+ * @run testng/othervm IdleConnectionTimeoutTest
+ */
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.net.http.HttpClient.Version.HTTP_2;
+import static org.testng.Assert.assertEquals;
+
+public class IdleConnectionTimeoutTest {
+
+    Http2TestServer http2TestServer;
+    URI timeoutUri;
+    URI noTimeoutUri;
+    final String IDLE_CONN_PROPERTY = "jdk.httpclient.idleConnectionTimeout";
+    final String TIMEOUT_PATH = "/serverTimeoutHandler";
+    final String NO_TIMEOUT_PATH = "/noServerTimeoutHandler";
+    static boolean expectTimeout;
+    static final PrintStream testLog = System.err;
+
+    @BeforeTest
+    public void setup() throws Exception {
+        http2TestServer = new Http2TestServer(false, 0);
+        http2TestServer.addHandler(new ServerTimeoutHandler(), TIMEOUT_PATH);
+        http2TestServer.addHandler(new ServerNoTimeoutHandler(), NO_TIMEOUT_PATH);
+
+        http2TestServer.start();
+        int port = http2TestServer.getAddress().getPort();
+        timeoutUri = new URI("http://localhost:" + port + TIMEOUT_PATH);
+        noTimeoutUri = new URI("http://localhost:" + port + NO_TIMEOUT_PATH);
+    }
+
+    /*
+       If the InetSocketAddress of the first remote connection is not equal to the address of the
+       second remote connection, then the idleConnectionTimeoutEvent has occurred and a new connection
+       was made to carry out the second request by the client. Otherwise, the old connection was reused.
+    */
+    @Test
+    public void testTimeoutFires() throws InterruptedException {
+        expectTimeout = true;
+        System.setProperty(IDLE_CONN_PROPERTY, "100");
+        HttpClient hc = HttpClient.newBuilder().version(HTTP_2).build();
+        HttpRequest hreq = HttpRequest.newBuilder(timeoutUri).version(HTTP_2).GET().build();
+
+        CompletableFuture<HttpResponse<String>> request = hc.sendAsync(hreq, HttpResponse.BodyHandlers.ofString(UTF_8));
+        HttpResponse<String> hresp = request.join();
+        assertEquals(hresp.statusCode(), 200);
+        // Sleep for 4x the timeout value to ensure that it occurs
+        Thread.sleep(800);
```

Comment:
Should this use the `Utils.adjustTimeout` to take into account the `TIMEOUT_FACTOR`?

Reply:
I don't think so? We're not waiting on some computational thing here - everybody should be sleeping ;-)

Reply:
You are right indeed - the `TIMEOUT_FACTOR` shouldn't play a role here.